src/yaml/to_yaml.cr

```diff
@@ -51,69 +59,95 @@ struct NamedTuple
 end
 
 class String
-  def to_yaml(yaml : YAML::Builder)
-    yaml.scalar self
+  def to_yaml(yaml : YAML::Nodes::Builder)
+    if YAML::Schema::Core.reserved_string?(self)
+      yaml.scalar self, style: YAML::ScalarStyle::DOUBLE_QUOTED
+    else
+      yaml.scalar self
+    end
   end
 end
 
 struct Number
-  def to_yaml(yaml : YAML::Builder)
-    yaml.scalar self
+  def to_yaml(yaml : YAML::Nodes::Builder)
+    yaml.scalar self.to_s
   end
 end
 
 struct Nil
-  def to_yaml(yaml : YAML::Builder)
+  def to_yaml(yaml : YAML::Nodes::Builder)
     yaml.scalar ""
   end
 end
 
 struct Bool
-  def to_yaml(yaml : YAML::Builder)
+  def to_yaml(yaml : YAML::Nodes::Builder)
     yaml.scalar self
   end
 end
 
 struct Set
-  def to_yaml(yaml : YAML::Builder)
+  def to_yaml(yaml : YAML::Nodes::Builder)
     yaml.sequence do
       each &.to_yaml(yaml)
     end
   end
 end
 
 struct Symbol
-  def to_yaml(yaml : YAML::Builder)
+  def to_yaml(yaml : YAML::Nodes::Builder)
     yaml.scalar self
   end
 end
 
 struct Enum
-  def to_yaml(yaml : YAML::Builder)
+  def to_yaml(yaml : YAML::Nodes::Builder)
     yaml.scalar value
   end
 end
 
 struct Time
-  def to_yaml(yaml : YAML::Builder)
-    yaml.scalar Time::Format::ISO_8601_DATE_TIME.format(self)
+  def to_yaml(yaml : YAML::Nodes::Builder)
+    if kind.utc? || kind.unspecified?
```

Comment:
Can't this be encapsulated in a format somehow? It seems pretty useful :-)

Reply:
Yes! We can probably do that later. Also, right now just 3 decimals are shown but after we merge the PR that makes Time hold nanoseconds we can show all 9 digits.